### DISCUSSION TITLE
Question About SSR and Tailwind CSS in Next.js

### DISCUSSION THREAD

**Discussion by mohamad-fallah** [tailwindlabs/tailwindcss#15295] [Help]
Hello Tailwind CSS team,

I have a question regarding how Tailwind CSS can be used in a Server-Side Rendering (SSR) environment. Specifically, I want to know if it's possible for Tailwind CSS to be built on the server and then sent to the browser as part of the SSR process.

Is it possible to generate the Tailwind CSS on the server during SSR, so that it can be delivered to the browser?
How does Tailwind CSS handle CSS generation in SSR environments? Does it require specific configurations to properly generate the CSS on the server?
Looking forward to your guidance and best practices on how to set this up properly!

Best regards,

**Comment by alex-eri** [comment]
I found a way to generate it on build time  with vite. I found a way to generate it in browser with `@tailwindcss/browser`. But I need ssr time generation because classes loaded from database not provided at build time.
